test/models/stacks_test.rb

```diff
@@ -263,6 +272,16 @@ def setup
       assert @stack.active_task?
     end
 
+    test ".run_deploy_in_foreground triggers a deploy" do
+      stack = Stack.create!(repo_owner: 'foo', repo_name: 'bar', environment: 'production')
+      commit = shipit_commits(:first)
+      stack.commits << commit
+
+      Stack.any_instance.stubs(:trigger_deploy).with(anything, anything, env: [], force: true, run_now: true)
```

Comment:
(reposting comment on new diff)

I believe that with the stubs.with, it'll only fail if the method was called with the wrong arguments. Vs. if you go with expects.with, it'll fail if the method was not called at all, or if it was called with the wrong arguments

Reply:
I misunderstood you last time. I'll try it out. Thanks for clarifying it :)